src/Services/Exceptions/NotSearchableException.php

```diff
@@ -0,0 +1,7 @@
+<?php
+
+namespace Algolia\Settings\Services\Exceptions;
+
+final class NotSearchableException extends \Exception
```

Comment:
If we introduce this exception with the public `validateClassSearchable` people will start using it for a different usecase. I'd rather keep the foot print of the extension as small as possible. WDYT?

Reply:
As small as possible definitely makes sense to me. Also, even after many years I still haven't made up my mind about using Exceptions for failed validation. So I am not hooked on introducing new Exceptions.

Reply:
Same here, totally agree with Exceptions.